lib/plurimath/math/function/unary_function.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+module Plurimath
+  module Math
+    module Function
+      class UnaryFunction
+        def ==(object)
+          object.parameter_value == parameter_value
+        end
+
+        def parameter_value
+          instance_variable_get(instance_variables[0])
```

Comment:
@suleman-uzair is `instance_variables` the default Ruby `Object.instance_variables`? This seems to be a strange usage of it that can be problematic.

Reply:
> @suleman-uzair is `instance_variables` the default Ruby `Object.instance_variables`? This seems to be a strange usage of it that can be problematic.

Yes, you are right, it can be done in a better way. I’ve changed it and updated the code.